### ISSUE TITLE
Can't access request data and resend the same request when random payload were used

### ISSUE THREAD

**Issue by onmywaytoheaven** [rangav/thunder-client-support#941]
**Describe the bug**
I can't access request data and resend the same request when I use random payload via custom filter.

For example. I created custom filter for random payload generation in order to assess proper backend validation. However, after I got failed iteration and opened failed request, I can't find the data that was submitted, all I see is variable with filter definition.
![image](https://user-images.githubusercontent.com/43062804/209311379-752feb54-67c2-485d-8d29-2b8b2e2abe29.png)

If I tried to get Code Snippet of request, it will generate random payload everytime I click on it.

**To Reproduce**
1. Use custom filter to generate random payload
```
async function fuzzFilter() {
    var _ = await tc.loadModule("underscore");
    var utf8 = await tc.loadModule("utf8");
    randomString = function (options) {
        var charCodes, charCodeLimit;
    
        options = options || {};
        charCodes = [];
        charCodeLimit = Math.pow(2, 16);
    
        _.times(_.random(options.maximumLength || 10), function () {
          charCodes.push(_.random(charCodeLimit));
        });
    
        return String.fromCharCode.apply(null, charCodes);
    };
    var res = randomString();
    return utf8.encode(res);
}
module.exports = [fuzzFilter];
```
2. Run request with fuzzFilter for ex. in header

**Expected behavior**
I can find exact request and resend it :)

**Platform:** 
 - OS: Windows 10 22H2
 - vscode version: 1.74.1
 - extension version: 2.2.0

**Your Team Size Using TC:**
Just me

**Comment by rangav** [comment]
Thanks @onmywaytoheaven for reporting the bug,

just to clarify- the request you send is not saved in the `sidebar?`

**Comment by rangav** [comment]
Can you see any errors in logs?

**Comment by onmywaytoheaven** [comment]
I am running it from collection.
In Thunder Client output I can see request headers for example, but it will be tough to find one when I run thousand iterations of the collection

**Comment by rangav** [comment]
so you want to see request details send in `logs `for a request that is executed from Run Collection right?

you can re-run that single request see in logs from Run collection view
![image](https://user-images.githubusercontent.com/8637550/209324533-ec12a9f9-7472-4306-a4f2-7e347b6954dc.png)

**Comment by onmywaytoheaven** [comment]
Moreover, I find that sometimes custom filter is not working.
For example, 
```
function thisFilterDoesNotWork() {
    var charCodes = [
        2,
        118,
        178,
        149
    ]
    var resultedString = String.fromCharCode.apply(null, charCodes);
    console.log(resultedString);
    return resultedString;
}
```

If you will try to run the filter above, for ex, in header, your header will remain as `{{@ | thisFilterDoesNotWork}}`.
Thats because of charcode 2 in it. Deleting it will resolve a problem

**Comment by rangav** [comment]
Thanks for reporting, will verify the issue.

**Comment by onmywaytoheaven** [comment]
> so you want to see request details send in `logs `for a request that is executed from Run Collection right?
> 
> you can re-run that single request see in logs from Run collection view ![image](https://user-images.githubusercontent.com/8637550/209324533-ec12a9f9-7472-4306-a4f2-7e347b6954dc.png)

yes, I want to resend exactly the same request

when I have failed iteration and click on the failed request
![image](https://user-images.githubusercontent.com/43062804/209326129-f030df0a-6253-40cc-8dd2-53e5a36626ce.png)
it will open it in a new tab
![image](https://user-images.githubusercontent.com/43062804/209326202-fd129b53-c8f1-405b-b1fd-d2add7cc984a.png)
but I will not find exact request here, as you see, the value of Fuzz-Header is variable, not string

**Comment by rangav** [comment]
The header value will not be the same, as its a dynamic value.

**Comment by rangav** [comment]
what solution would you like here?

**Comment by onmywaytoheaven** [comment]
Yes, its dynamic, but when your test is failed, you need the same request to validate it, right? I guess its a bit counterintuitive that when I click on Run on the failed request, I get other request and other results.
My think that request can be accessed on the right pane, with response... And for ex. when you click on Run on the failed test, request should be populated with actual data, that were sended.

**Comment by rangav** [comment]
The purpose of dynamic value is to generate random value, and your tests needs to take this into account. When re-run a request it should still pass the all tests.

The solution you proposed will complicate things.

**Comment by onmywaytoheaven** [comment]
Yes, it should. But if request fails, I can't even know why, I can't send it again.

**Comment by rangav** [comment]
Well the `Results` tab will show why it failed and also you can `re-send` the request. both are possible as told you the above.

Can you share screenshot your request screenshot and which tests failed?

**Comment by onmywaytoheaven** [comment]
Its just a sandbox
![image](https://user-images.githubusercontent.com/43062804/209330756-94500c84-f11e-47b9-9df8-82f687ab3e9e.png)

Test: responseCode == 400

**Comment by onmywaytoheaven** [comment]
> Well the `Results` tab will show why it failed and also you can `re-send` the request. both are possible as told the above.
> 
> Can you share screenshot your request and which tests failed?

Its possible, but Results tab and rerunning the request would not help me to get the same result

**Comment by rangav** [comment]
from your screenshot, you know its failed because you 400 for responseCode, thats clear right?

Currently we dont `save` the dynamic values generated while sending request.

**Comment by onmywaytoheaven** [comment]
> from your screenshot, you know its failed because you 400 for responseCode, thats clear right?
> 
> Currently we dont `save` the dynamic values generated while sending request.

Sure, I know the outcomes, but I don't know what was the cause.
It will be great if you ever implement such feature 😊
Thank you for Thunder Client!

**Comment by rangav** [comment]
Sure will note your feedback, if we ever implement this feature will update here.

I look into the special characters issue your have with function - `thisFilterDoesNotWork`

**Comment by rangav** [comment]
> Moreover, I find that sometimes custom filter is not working. For example,
> 
> ```
> function thisFilterDoesNotWork() {
>     var charCodes = [
>         2,
>         118,
>         178,
>         149
>     ]
>     var resultedString = String.fromCharCode.apply(null, charCodes);
>     console.log(resultedString);
>     return resultedString;
> }
> ```
> 
> If you will try to run the filter above, for ex, in header, your header will remain as `{{@ | thisFilterDoesNotWork}}`. Thats because of charcode 2 in it. Deleting it will resolve a problem

You need to escape the `resultedString` for this to work. dont know how to escape this special char, try `JSON.stringify`
